File: backend/spring/src/main/java/io/ssafy/openticon/controller/response/PurchaseEmoticonResponseDto.java
package io.ssafy.openticon.controller.response;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.RequiredArgsConstructor;

import java.util.List;

@Getter
@AllArgsConstructor
@Builder
public class PurchaseEmoticonResponseDto {
    private Long packId;

    private String packName;

    private String thumbnailImg;

    private Boolean isHide;

    private Boolean isPublic;

    private String listImage;

    private List<String> emoticons;
}

File: android/app/src/main/java/io/ssafy/openticon/data/model/PurchaseEmoticonResponseDto.kt
package io.ssafy.openticon.data.model

import io.ssafy.openticon.domain.model.EmoticonPack

data class PurchaseEmoticonResponseDto(
    val packId: Int,
    val packName: String,
    val thumbnailImg: String,
    val isHide: Boolean,
    val isPublic: Boolean,
    val listImage: String
)

fun PurchaseEmoticonResponseDto.toEmoticonPack(): EmoticonPack {
    return EmoticonPack(
        id = packId,
        title = packName,
        thumbnail = thumbnailImg,
        listImg = listImage,
        isPublic = isPublic
    )
}